pkg/ksync/cli.go

```diff
@@ -11,39 +11,58 @@ import (
 	"github.com/jenkins-x/jx/pkg/util"
 )
 
-// InstallKSync install ksync
-func InstallKSync() (string, error) {
+// KSyncer contains the methods used for ksync
+type KSyncer interface {
+	GetJXBinLocation() (string, error)
+	IsInstalled(string) (string, bool, error)
+	RunCommand(name string, args []string) (string, error)
+	GetTag(binDir, fileName, owner, repo string) (string, error)
+}
+
+// KsynClier implements the methods contained in KSyncer.
+type KsynClier struct{}
+
+var commit string
+
+// GetJXBinLocation returns the jx bin location
+func (kcli *KsynClier) GetJXBinLocation() (string, error) {
 	binDir, err := util.JXBinLocation()
 	if err != nil {
 		return "", err
 	}
-	binary := "ksync"
+	return binDir, nil
+}
+
+// IsInstalled checks if the binary is installed
+func (kcli *KsynClier) IsInstalled(binary string) (string, bool, error) {
 	fileName, flag, err := packages.ShouldInstallBinary(binary)
```

Comment:
Depending on what you try to achieve `ShouldInstallBinary` is not going to work for you. For example if your aim is to install ksync into JX_HOME/bin, then  `ShouldInstallBinary` is not a good choice. See also issue #6912

Reply:
I think I can remove `ShouldInstallBinary` after issue #6912 is resolved.